user/module/probe_openssl_lib.go

```diff
@@ -119,67 +118,63 @@ func (m *MOpenSSLProbe) detectOpenssl(soPath string) error {
 		return nil
 	}
 
-	sectionSize := int64(s.Offset)
+	sectionOffset := int64(s.Offset)
+	sectionSize := s.Size
+
+	r.Close()
 
 	_, err = f.Seek(0, 0)
 	if err != nil {
 		return err
 	}
 
-	ret, err := f.Seek(sectionSize, 0)
-	if ret != sectionSize || err != nil {
+	ret, err := f.Seek(sectionOffset, 0)
+	if ret != sectionOffset || err != nil {
 		return err
 	}
 
-	buf := make([]byte, s.Size)
-	if buf == nil {
-		return nil
-	}
+	versionKey := ""
 
-	_, err = f.Read(buf)
+	// e.g : OpenSSL 1.1.1j  16 Feb 2021
+	rex, err := regexp.Compile(`(OpenSSL\s\d\.\d\.[0-9a-z]+)`)
 	if err != nil {
-		return err
-	}
-
-	// 按照\x00 拆分  buf
-	var slice [][]byte
-	if slice = bytes.Split(buf, []byte("\x00")); slice == nil {
 		return nil
 	}
 
-	dumpStrings := make(map[uint64][]byte, len(slice))
-	length := uint64(len(slice))
+	buf := make([]byte, 1024*1024) // 1Mb
+	totalReadCount := 0
+	for totalReadCount < int(sectionSize) {
```

Comment:
If the string "OpenSSL 1.1.1j 16 Feb 2021" is split into two buffers, will it be missed?